### PR TITLE
feat(kit): `Time` supports `SS.MSS` & `MM.SS.MSS` modes

### PR DESCRIPTION
**Pull request by KrollikRoddzer**
Closes https://github.com/taiga-family/maskito/issues/38

### CODE REVIEW

File: projects/kit/src/lib/utils/time/parse-time-string.ts [hunk 1]
```diff
@@ -3,17 +3,38 @@ import type {MaskitoTimeSegments} from '../../types';
 /**
  * @param timeString can be with/without fixed characters
  */
-export function parseTimeString(timeString: string): Partial<MaskitoTimeSegments> {
+export function parseTimeString(
```
[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/kit/src/lib/utils/time/parse-time-string.ts:6]:
```ts
import {describe, expect, it} from '@jest/globals';

import {parseTimeString} from '../parse-time-string';

describe('parseTimeString', () => {
    it('HH:MM', () => {
        expect(parseTimeString('23.59', 'HH:MM')).toEqual({
            hours: '23',
            minutes: '59',
        });
    });

    it('SS.MSS', () => {
        expect(parseTimeString('59.999', 'SS.MSS')).toEqual({
            seconds: '59',
            milliseconds: '999',
        });
    });

    // [all other time modes tests]
});

```

File: projects/kit/src/lib/utils/time/parse-time-string.ts [hunk 2]
```diff
@@ -3,17 +3,38 @@ import type {MaskitoTimeSegments} from '../../types';
...
+        Number(timeMode.startsWith('SS')) * 4 +
+        Number(timeMode.startsWith('MSS')) * 6;
+
+    const changeSelection = (index: number): number => Math.max(index - offset, 0);
+
+    return {
+        hours: [changeSelection(0), changeSelection(2)],
+        minutes: [changeSelection(2), changeSelection(4)],
+        seconds: [changeSelection(4), changeSelection(6)],
+        milliseconds: [changeSelection(6), changeSelection(9)],
```
Comment (nsbarsukov) [inline projects/kit/src/lib/utils/time/parse-time-string.ts:38]: What do you think about this solution?

```ts
import type {MaskitoTimeMode, MaskitoTimeSegments} from '../../types';

const SEGMENT_FULL_NAME: Record<string, keyof MaskitoTimeSegments> = {
    HH: 'hours',
    MM: 'minutes',
    SS: 'seconds',
    MSS: 'milliseconds',
};

/**
 * @param timeString can be with/without fixed characters
 */
export function parseTimeString(
    timeString: string,
    timeMode: MaskitoTimeMode,
): Partial<MaskitoTimeSegments> {
    const onlyDigits = timeString.replaceAll(/\D+/g, '');

    let offset = 0;

    return Object.fromEntries(
        timeMode.split(/\W/).map(segmentAbbr => {
            const segmentValue = onlyDigits.slice(offset, offset + segmentAbbr.length);

            offset += segmentAbbr.length;

            return [SEGMENT_FULL_NAME[segmentAbbr], segmentValue];
        }),
    );
}

```

File: projects/kit/src/lib/utils/time/to-time-string.ts [hunk 3]
```diff
@@ -6,9 +6,10 @@ export function toTimeString({
...
 }: Partial<MaskitoTimeSegments>): string {
-    const mm = minutes && `:${minutes}`;
-    const ss = seconds && `:${seconds}`;
-    const ms = milliseconds && `.${milliseconds}`;
+    const mm = hours ? minutes && `:${minutes}` : minutes;
+    const ss = hours || minutes ? seconds && `:${seconds}` : seconds;
+    const ms =
+        hours || minutes || seconds ? milliseconds && `.${milliseconds}` : milliseconds;
 
     return `${hours}${mm}${ss}${ms}`;
```
Comment (nsbarsukov) [inline projects/kit/src/lib/utils/time/to-time-string.ts:14]: What do you think about this solution ?

```ts
import type {MaskitoTimeSegments} from '../../types';

const LEADING_NON_DIGITS = /^\D*/;
const TRAILING_NON_DIGITS = /\D*$/;

export function toTimeString({
    hours = '',
    minutes = '',
    seconds = '',
    milliseconds = '',
}: Partial<MaskitoTimeSegments>): string {
    return `${hours}:${minutes}:${seconds}.${milliseconds}`
        .replace(LEADING_NON_DIGITS, '')
        .replace(TRAILING_NON_DIGITS, '');
}
```

File: projects/kit/src/lib/utils/time/to-time-string.ts [hunk 4]
```diff
@@ -6,9 +6,10 @@ export function toTimeString({
     seconds = '',
     milliseconds = '',
 }: Partial<MaskitoTimeSegments>): string {
-    const mm = minutes && `:${minutes}`;
-    const ss = seconds && `:${seconds}`;
-    const ms = milliseconds && `.${milliseconds}`;
+    const mm = hours ? minutes && `:${minutes}` : minutes;
+    const ss = hours || minutes ? seconds && `:${seconds}` : seconds;
+    const ms =
+        hours || minutes || seconds ? milliseconds && `.${milliseconds}` : milliseconds;
```
Comment (nsbarsukov) [inline projects/kit/src/lib/utils/time/to-time-string.ts:12]: Add unit tests for `toTimeString` utility, please

```ts
import {describe, expect, it} from '@jest/globals';

import {toTimeString} from '../to-time-string';

describe('toTimeString', () => {
    it('MM.SS.MSS', () => {
        expect(
            toTimeString({
                minutes: '59',
                seconds: '30',
                milliseconds: '123',
            }),
        ).toBe('59:30.123');
    });

    it('SS.MSS', () => {
        expect(
            toTimeString({
                seconds: '59',
                milliseconds: '123',
            }),
        ).toBe('59.123');
    });

    // [all other mode tests]
});

```

File: projects/kit/src/lib/utils/time/parse-time-string.ts (hunk 2 above)
Comment (KrollikRoddzer) [reply projects/kit/src/lib/utils/time/parse-time-string.ts:38]: I think it's better. It uses easier logic

File: projects/kit/src/lib/utils/time/to-time-string.ts (hunk 3 above)
Comment (KrollikRoddzer) [reply projects/kit/src/lib/utils/time/to-time-string.ts:14]: It's easier. I'll use it

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**